pkg/selector/comparators.go

```diff
@@ -219,6 +265,16 @@ func getEBSOptimizedBaselineIOPS(ebsInfo *ec2.EbsInfo) *int64 {
 	return ebsInfo.EbsOptimizedInfo.BaselineIops
 }
 
+func getCPUManufacturer(instanceTypeInfo *ec2.InstanceTypeInfo) *string {
+	if contains(instanceTypeInfo.ProcessorInfo.SupportedArchitectures, ec2.ArchitectureTypeArm64) {
+		return aws.String("aws")
+	}
+	if amdRegex.Match([]byte(*instanceTypeInfo.InstanceType)) {
```

Comment:
Regex matching the instance type seems fragile with all the possible current and future instance type names. Is that really the best (or only) way to suss out if the CPU is AMD?

Reply:
It is currently the only way to tell if a node is AMD :(  But it's well documented in instance type names https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#:~:text=capabilities%2C%20such%20as%3A-,a%20%E2%80%93%20AMD%20processors,-g%20%E2%80%93%20AWS%20Graviton